cloudpathlib/cloudpath.py

```diff
@@ -535,20 +532,43 @@ def _dispatch_to_path(self, func, *args, **kwargs):
                 self._new_cloudpath(_resolve(p)) for p in path_version if _resolve(p) != p.root
             )
 
-        # when pathlib something else, we probably just want that thing
+        # when pathlib returns something else, we probably just want that thing
         # cases this should include: str, empty sequence, sequence of str, ...
         else:
             return path_version
 
     def __truediv__(self, other):
-        if not isinstance(other, (str,)):
+        if not isinstance(other, (str, PurePosixPath)):
             raise TypeError(f"Can only join path {repr(self)} with strings.")
 
         return self._dispatch_to_path("__truediv__", other)
 
     def joinpath(self, *args):
         return self._dispatch_to_path("joinpath", *args)
 
+    def absolute(self):
+        return self
+
+    def is_absolute(self):
+        return True
+
+    def resolve(self, strict=False):
+        return self
+
+    def relative_to(self, other):
+        # We don't dispatch regularly since this never returns a cloud path (since it is relative, and cloud paths are
+        # absolute)
+        if not isinstance(other, CloudPath):
```

Comment:
* AFAICT, the buckets/containers are included in `_path`, so this should already work - did you mean a different error message is needed?
 * Good catch, I assumed the cloud prefix is included in `_path` but I see it isn't. I fixed it. Not sure how to test this however. Is there a rig for testing different provider cloud paths somewhere? Or should I set something up?
 * I'd argue it's not intuitive for client instances to play a part in `relative_to`. It might even cause issues when using `LocalXPath` for testing. I think treating them as fancy strings makes the most sense. I'm not familiar with Azure though.

Reply:
> AFAICT, the buckets/containers are included in _path, so this should already work - did you mean a different error message is needed?

Yep, you're right. This is good as implemented.

> Good catch, I assumed the cloud prefix is included in _path but I see it isn't. I fixed it. Not sure how to test this however. Is there a rig for testing different provider cloud paths somewhere? Or should I set something up?

Fix here looks good. In terms of testing, you could just do something like:

```python
p = rig.create_cloud_path(...)
new_prefix = "az://" if not p.clout_prefix == "az://" else "s3://"
not_p = CloudPath(new_prefix + p._no_prefix)
```

> I'd argue it's not intuitive for client instances to play a part in relative_to. It might even cause issues when using LocalXPath for testing. I think treating them as fancy strings makes the most sense. I'm not familiar with Azure though.

Seems reasonable to me!